src/v/transform/transform_processor.cc

```diff
@@ -217,13 +217,20 @@ processor::load_latest_committed() {
                       vlog(_logger.debug, "starting at latest: {}", latest);
                       return ssx::now(latest);
                   },
-                  [this](model::timestamp ts) {
+                  [this, &latest](model::timestamp ts) {
                       vlog(_logger.debug, "starting at timestamp: {}", ts);
                       // We want to *start at* this timestamp, so record that
                       // we're going to commit progress at the offset before, so
-                      // we start inclusive of this offset.
+                      // we start inclusive of this offset. If nothing has been
+                      // committed since the start timestamp, commit progress at
+                      // latest (i.e. start from the end)
                       return _source->offset_at_timestamp(ts, &_as).then(
-                        kafka::prev_offset);
+                        [&latest](
+                          kafka::offset o) -> ss::future<kafka::offset> {
+                            return ssx::now(
+                              o != kafka::offset::min() ? kafka::prev_offset(o)
```

Comment:
We should cherry-pick fix this into 24.1, and also can we just return the nullopt in the API then? I misunderstood nullopt to only be returned when there are no records in the topic.

Reply:
Or just return latest offset here: https://github.com/redpanda-data/redpanda/blob/92a4a698d12cdf645c8ddb4bfc46bb787cadfe52/src/v/transform/api.cc#L161

Reply:
>cherry pick

oh yeah, good point. will split off into a separate PR.

> return nullopt in the API...or just return latest

Sure. Took me a bit of experimentation to discern exactly what the issue was here, and I wasn't 100% confident that the min was strictly not needed in all cases. Can litigate on the PR tho.

Reply:
https://github.com/redpanda-data/redpanda/pull/20127